pkg/providers/vsphere/vsphere_test.go

```diff
@@ -3245,6 +3245,68 @@ func TestClusterSpecChangedMachineConfigsChanged(t *testing.T) {
 	}
 }
 
+func TestValidateMachineConfigsNameUniquenessSuccess(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
```

Comment:
can we reuse `newProviderTest()` instead of setting all test components one by one?